telemetry/telemetrybuffer.go

```diff
@@ -229,8 +229,8 @@ func (tb *TelemetryBuffer) PushData(ctx context.Context) {
 }
 
 // read - read from the file descriptor
-func read(conn net.Conn) (b []byte, err error) {
-	b, err = bufio.NewReader(conn).ReadBytes(Delimiter)
+func read(reader *bufio.Reader) (b []byte, err error) {
```

Comment:
can you just inline this? i don't see a need for it to be a helper and separating it like this probably caused the issue